apis/storage/v1alpha3/types.go

```diff
@@ -58,9 +58,20 @@ func CopyToProjectTeam(pt *ProjectTeam) *storage.ProjectTeam {
 // ACLRule represents a grant for a role to an entity (user, group or team) for a
 // Google Cloud Storage object or bucket.
 type ACLRule struct {
-	Entity      string       `json:"entity,omitempty"`
+	// Entity refers to a user or group. They are sometimes referred to as grantees.
+	// It could be in the form of:
+	// "user-<userId>", "user-<email>", "group-<groupId>", "group-<email>",
+	// "domain-<domain>" and "project-team-<projectId>".
+	//
+	// Or one of the predefined constants: AllUsers, AllAuthenticatedUsers.
+	Entity string `json:"entity,omitempty"`
+
+	// Role is the access permission for the entity.
+	// Valid values are "OWNER", "READER" and "WRITER"
+	// +kubebuilder:validation:Enum=OWNER;READER;WRITER
+	Role string `json:"role,omitempty"`
+
 	EntityID    string       `json:"entityId,omitempty"`
-	Role        string       `json:"role,omitempty"`
 	Domain      string       `json:"domain,omitempty"`
```

Comment:
For other fields, I couldn't find any documentation in SDK source code and the publich REST API documentation here says you use `Role` and `Entity` field https://cloud.google.com/storage/docs/json_api/v1/bucketAccessControls/insert

Reply:
why is this type called `ACLRule` if the external resource is called `BucketAccessControls`? Are they the same thing? 

There is some short/quick documentation for the other fields such as entityID and domain in https://cloud.google.com/storage/docs/json_api/v1/bucketAccessControls#resource. Are those useful to add here?

Reply:
As far as I see, there are two ways to manage `BucketAccessControls`: inline and as a separate resource (similar to GKECluster<>NodePool). In the current implementation, we use the inline notation, in which the name is `ACL`. Take a look at https://cloud.google.com/storage/docs/json_api/v1/buckets/insert and that's the actual SDK object we create:
```
// BucketAttrs represents the metadata for a Google Cloud Storage bucket.
// Read-only fields are ignored by BucketHandle.Create.
type BucketAttrs struct {
	// Name is the name of the bucket.
	// This field is read-only.
	Name string

	// ACL is the list of access control rules on the bucket.
	ACL []ACLRule
......
```
When we update this resource to be `v1beta1`, we'll need to manage the `BucketAccessControl` object separately and the name of the resource will match.

> There is some short/quick documentation for the other fields such as entityID and domain

Updated the rest of the comments with the documentation you linked.